Fixed #32732 -- Removed usage of deprecated 'db' and 'passwd' connection options in MySQL backend.

pope1ni opened the pull request:
ticket-32732

The `db` and `passwd` connection arguments have been deprecated and `database` and `password`, available since `mysqlclient>=1.3.8` should be used instead.

felixxm commented on django/db/backends/mysql/client.py:
```diff
@@ -8,11 +8,11 @@ class DatabaseClient(BaseDatabaseClient):
         password = settings_dict['OPTIONS'].get(
             'password',
-            settings_dict['OPTIONS'].get('passwd', settings_dict['PASSWORD'])
+            settings_dict['OPTIONS'].get('password', settings_dict['PASSWORD'])
```
This change is unnecessary and unrelated. We support both `password` and `passwd` options in `DATABASES['OPTIONS']`.

felixxm commented on django/db/backends/mysql/client.py:
```diff
@@ -8,11 +8,11 @@ class DatabaseClient(BaseDatabaseClient):
         args = [cls.executable_name]
         env = None
-        db = settings_dict['OPTIONS'].get('db', settings_dict['NAME'])
+        database = settings_dict['OPTIONS'].get('database', settings_dict['NAME'])
```
We should support both `db` and `database`, e.g.
```python
database = settings_dict['OPTIONS'].get(
    'database',
    settings_dict['OPTIONS'].get('db', settings_dict['NAME']),
)
```

pope1ni replied:
🤦🏻‍♂️ Yes.

felixxm reviewed (commented):
@pope1ni Thanks for this patch :+1: Please add a separate test for the `database` option in ` DATABASES['OPTIONS']`, and small release notes, e.g.
```
* Prepared for ``mysqlclient`` > 2.0.3 support (:ticket:`32732`).
```

pope1ni commented:
Thanks @felixxm

> Please add a separate test for the `database` option in ` DATABASES['OPTIONS']`

I updated `test_options_override_settings_proper_values` to test with both `database`/`password` and `db`/`passwd`. This made `test_options_password` obsolete. I also added `test_options_non_deprecated_keys_preferred` to ensure that `database`/`password` are preferred over `db`/`passwd`.

felixxm commented:
@pope1ni Thanks for updates :+1:
